Fix crash in PlaybackInfo.calculateTotalDuration, #5095

low-batt opened the pull request:
This commit will:
- Add the `Atomic` property wrapper to `PlaybackInfo.playlist`
- Remove `PlaybackInfo.lockQueue` and instead use the playlist lock to single thread access to the playlist itself as well as the playlist cache
- Change all code that was directly accessing the playlist to use the lock

- [x] I have read [CONTRIBUTING.md](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md)
- [x] This implements/fixes issue #5095.

---

**Description:**

low-batt committed:
Fix crash in PlaybackInfo.calculateTotalDuration, #5095

This commit will:
- Add the Atomic property wrapper to PlaybackInfo.playlist
- Remove PlaybackInfo.lockQueue and instead use the playlist lock to
  single thread access to the playlist itself as well as the playlist
  cache
- Change all code that was directly accessing the playlist to use the
  lock

uiryuu commented on iina/PlaylistViewController.swift:
```diff
@@ -644,9 +650,13 @@ class PlaylistViewController: NSViewController, NSTableViewDataSource, NSTableVi
-    let files = selectedRows!.enumerated().map { (_, i) in
-      URL(fileURLWithPath: player.info.playlist[i].filename)
-    }
+
```
Extra line?

low-batt replied:
Fixed!

uiryuu reviewed (approved):
Nice and clean changes to me

low-batt committed:
Fix crash in PlaybackInfo.calculateTotalDuration, #5095

This commit will:
- Add the Atomic property wrapper to PlaybackInfo.playlist
- Remove PlaybackInfo.lockQueue and instead use the playlist lock to
  single thread access to the playlist itself as well as the playlist
  cache
- Change all code that was directly accessing the playlist to use the
  lock
